### PR TITLE
Replace ToxDNS with DHT level name queries.

### PR DESCRIPTION
**Pull request by GrayHatter**

### CODE REVIEW

File: toxcore/query.c [hunk 1]
```diff
@@ -0,0 +1,190 @@
+/** query.c
+ *
+ * Makes requests for information using the DHT or Onion as approitate.
+ *
+ * This file should include a licence, but if you're reading this, it doesn't yet so everything in
+ * this file is donated to both tox, toktok, and you under a BSD or MIT licence at any option.
```
Comment (zetok) [inline toxcore/query.c]: Licensing doesn't work like that. This is licensed under GPL, do it properly. Same goes for the .h file.

Comment (SkyzohKey) [inline toxcore/query.c]: @zetok GrayHatter can do whatever he want's with his code, IMHO. I don't see why this couldn't works like that

**Comment by GrayHatter** [comment]
@zetok, it actually does work like that. Until I submit it to and it's
accepted by I hold total licensing rights. At that point, I can use any GPL
compatible licence. And toktok can relicense how ever they'd like.

**Comment by zetok** [comment]
@GrayHatter bullshit. https://www.gnu.org/copyleft/gpl.html ← read it, since you sound like you have no idea what you're talking about.

**Comment by iphydf** [comment]
@zetok this is not how we have civilised discussions. "bullshit" has no place here. Please adapt your conversation style to be more respectful going forward.

@GrayHatter you have built a prototype now. Please write a design proposal as described in https://github.com/TokTok/jvm-toxcore-c/blob/master/doc/development.md.

I am closing this PR, as the code is not ready to be merged and a design does not yet exist.

**Pull request closed without merging**